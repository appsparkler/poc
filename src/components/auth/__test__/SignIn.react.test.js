import React from 'react'
import {SignIn} from '../SignIn.react'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, {shallow} from 'enzyme'

Enzyme.configure({adapter: new Adapter()})

it('should render correctly', ()=> {
  expect.assertions(1)

  const component = renderer.create(<SignIn />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})

it('update the state on input in email-field in form', ()=> {
  const component = shallow(<SignIn />)
  const emailInput = component.find('input#email')
  const CHANGED_VALUE = 'abc@def.com'
  emailInput.simulate('change', {
    target: {
      value: CHANGED_VALUE,
      id: 'email',
    },
  })
  expect(component.state().email).toEqual(CHANGED_VALUE)
})

it('update the state on-change in password field in sign-in-form', ()=> {
  const component = shallow(<SignIn />)
  const emailInput = component.find('input#password')
  const CHANGED_VALUE = 'jalksdjflkajdsf'
  emailInput.simulate('change', {
    target: {
      value: CHANGED_VALUE,
      id: 'password',
    },
  })
  expect(component.state().password).toEqual(CHANGED_VALUE)
})

it(`preventDefault and stopPropagation
    events to have been called on form-submit`, ()=> {
  expect.assertions(2)
  const component = shallow(<SignIn />)
  const emailInput = component.find('form')
  const preventDefault = jest.fn()
  const stopPropagation = jest.fn()
  emailInput.simulate('submit', {
    preventDefault: preventDefault,
    stopPropagation: stopPropagation,
  })
  expect(preventDefault).toHaveBeenCalledTimes(1)
  expect(stopPropagation).toHaveBeenCalledTimes(1)
})
