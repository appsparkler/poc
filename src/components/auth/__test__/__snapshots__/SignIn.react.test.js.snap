// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h4
    className="header grey-text text-darken-3"
  >
    Login
  </h4>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      className="pink-text"
      id="email"
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      className="pink-text"
      id="password"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="input-field"
  >
    <button
      className="btn pink lighten-1 z-depth-0"
      type="submit"
    >
      Login
    </button>
  </div>
</form>
`;
